repl/src/main/spark_2.2/org/apache/spark/repl/h2o/SparkSpecificUtils.scala

```diff
@@ -0,0 +1,46 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.repl.h2o
+
+import java.io.File
+
+import org.apache.spark.SparkConf
+import org.apache.spark.util.Utils
+
+object SparkSpecificUtils extends CrossSparkUtils {
+  override def getJars(conf: SparkConf): String = {
+
+    import scala.reflect.runtime.{universe => ru}
+    val instanceMirror = ru.runtimeMirror(this.getClass.getClassLoader).reflect(Utils)
+    val methodSymbol = ru.typeOf[Utils.type].decl(ru.TermName("getLocalUserJarsForShell"))
+    val jars = if (methodSymbol.isMethod) {
+      val method = instanceMirror.reflectMethod(methodSymbol.asMethod)
+      method(conf).asInstanceOf[Seq[String]]
+    } else {
+      // Fallback to Spark 2.2.0
```

Comment:
Is this `if-else` branch specific only to 2.2.0 and the first one to other Spark 2.2.x versions?

Reply:
Yup, it is. I will make it more specific in the comments, thanks!